### PR TITLE
Load balancer networking

### PR DESCRIPTION
**Pull request by bronaghs**
In order to determine which Floating Ips are available for recycling during instance provisioning, we need to consider that some may be associated with Load balancers, not just instances.

This PR associates a Floating Ip with its Load balancer through a fake network port; a necessary step given that one is not already present.

https://bugzilla.redhat.com/show_bug.cgi?id=1380728

### CODE REVIEW

**Comment by bronaghs** [comment]
@miq-bot add_label wip

File: app/models/manageiq/providers/azure/network_manager/floating_ip.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 class ManageIQ::Providers::Azure::NetworkManager::FloatingIp < ::FloatingIp
+  belongs_to :load_balancer
+
+  def self.available
+    where(:vm_id => nil, :load_balancer_id => nil)
```
Comment (Ladas) [inline app/models/manageiq/providers/azure/network_manager/floating_ip.rb]: does Azure document somewhere what entities can have a floating IP?

Comment (bronaghs) [inline app/models/manageiq/providers/azure/network_manager/floating_ip.rb]: Hey @ladas - https://msdn.microsoft.com/en-us/library/azure/mt163638.aspx - see paragraph 2

Comment (Ladas) [inline app/models/manageiq/providers/azure/network_manager/floating_ip.rb]: right, seems like only LB and VM can be associated here, so this should be good for now

**Comment by Ladas** [comment]
@bronaghs this should be good for now :+1: 

I am thinking, for cross providers, it could be enough to store the connected device as a text (e.g. device id column) and base 'available?' on that being nil?

Then this should work for all providers and also for objects we don't track yet, since we will see there is some id, but we can't link it in our DB.

**Comment by Ladas** [comment]
@miq-bot assign @blomquisg

**Comment by Ladas** [comment]
@blomquisg we can merge this, to tackle the current issue. Then as a follow up, we can see if there is a cross-provider solution.

**Comment by bronaghs** [comment]
@Ladas - can you take a look while I write the specs?

**Comment by djberg96** [comment]
@Ladas @bronaghs Sorry about the conflicting refresher.yml, @durandom had to regenerate the cassettes for one of his PR's. In theory you should be able to just use the new one.

**Comment by bronaghs** [comment]
@Ladas - please review

**Comment by Ladas** [comment]
@bronaghs looks great

**Pull request merged**